### DISCUSSION TITLE
LinkerD compatibility with EKS Fargate

### DISCUSSION THREAD

**Discussion by VladimirAkopyan** [linkerd/linkerd2#5535] [Q&A]
AWS and Azure offer 'serverless' implementations of Kubernetes, with [EKS Fargate](https://docs.aws.amazon.com/eks/latest/userguide/fargate.html) and [Azure Container Instances](https://azure.microsoft.com/en-gb/resources/videos/using-kubernetes-with-azure-container-instances/). These are popular because there is no maintenance burden with the VM, each pod appears to be hosted on it's own node. However deamonsets do not work, and this breaks many applications, for instance log collection, etc. 

I have used LinkerD previously, but only for experimentation. I have given it a quick spin on EKS Fargate and LinkerD would work with 'normal' VMs but not 'fargate' pods - you can mix them in the same cluster. 

Before I dive into the inner working of LinkerD, which I am not very knowledgeable in, I am wondering if this is a known supported or unsupported configuration? Has anyone worked with these types of platform before? I am happy to provide logs, etc. but if it's a knows unsupported configuration it would be great to know that.

**Comment by olix0r** [comment]
@VladimirAkopyan I think you may be the first person to try this ;) I don't know enough of the details of Fargate to know whether it should Just Work or whether changes may be needed. We'd be happy to help guide you, though.

**Comment by wmorgan** [comment]
AIUI Fargate does not provide the ability to either do NET_ADMIN or CNI chaining, so Linkerd will not work because it cannot change the pod iptables rules. Note that this is true of any service mesh (except AppMesh I assume).

But I could be wrong. I'm basing this on https://github.com/aws/containers-roadmap/issues/682 .
